torch/_inductor/sizevars.py

```diff
@@ -192,7 +192,12 @@ def _simplify_loops_impl(
         """
         sizes = list(map(self.simplify, sizes))
 
-        strides = [self.stride_vars(x, index_vars) for x in index_formulas]
+        strides = [
+            self.stride_vars(x, index_vars)
+            if isinstance(x, sympy.Expr)
+            else [0] * len(index_vars)
+            for x in index_formulas
```

Comment:
Can you leave a comment, or, even better, extend `stride_vars` to support bool exprs?

Reply:
Left a comment. Making `stride_vars` support boolean expressions doesn't really make sense. `s0 < 1023` will have a stride of 0 everywhere except at the boundary point, so it doesn't make sense to try and represent this with a stride.